webapp-django/crashstats/api/views.py

```diff
@@ -575,6 +577,89 @@ def finalize_response(self, request, response, *args, **kwargs):
         return response
 
 
+class MissingProcessedCrashAPI(SocorroAPIView):
+    """
+    API for retrieving list of missing processed crash reports.
+    """
+
+    API_NAME = "MissingProcessedCrash"
+
+    IS_PUBLIC = True
+
+    PAGE_SIZE = 1_000
+
+    HELP_TEXT = """
+    Retrieves list of missing processed crash report records. Each record denotes a
+    crash report that the system thought was missing. Some have been processed since
+    then.
+
+    :Method: HTTP GET
+    :Argument: (optional) ``page``: 1-based page to retrieve
+
+    Returns
+
+        {"results": [ RECORD+ ]}, "count": COUNT, "next": LINK, "previous": LINK}
+
+    where COUNT is the total number of records in the table, LINK is either null
+    or the url to a page, and RECORD is a structure like:
+
+        {"crash_id": CRASHID, "is_processed": BOOL, "created": TIMESTAMP}
+
+    """
+
+    @method_decorator(anonymous_csrf_exempt)
+    @method_decorator(csrf_exempt)
+    @method_decorator(
+        ratelimit(key="ip", method=["GET"], rate=utils.ratelimit_rate, block=True)
+    )
+    def get(self, request):
+        qs = models.MissingProcessedCrash.objects.order_by("id").all()
+        paginator = Paginator(qs, self.PAGE_SIZE)
+
+        try:
+            page_number = int(request.GET.get("page", "1"))
+        except ValueError:
+            return http.JsonResponse(
+                {"error": "invalid page number"}, status=status.HTTP_400_BAD_REQUEST
+            )
```

Comment:
We have a handful of ways to return responses and it's kind of weird. I think I looked at this a while back and decided to not do anything because we also have multiple "api frameworks" involved and they're pretty different.

Still, this doesn't thrill me.